core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -277,8 +277,8 @@ private[spark] class RestSubmissionClient(master: String) extends Logging {
   private def getBaseUrl(master: String): String = {
     var masterUrl = master
     supportedMasterPrefixes.foreach { prefix =>
-      if (master.startsWith(prefix)) {
-        masterUrl = master.stripPrefix(prefix)
+      if (masterUrl.startsWith(prefix)) {
+        masterUrl = masterUrl.stripPrefix(prefix)
```

Comment:
Yes, it seems that the issue is that we have something like `spark://mesos://`. Instead we should fix the place where we prepend it.